sdk/core/Azure.Core.Expressions.DataFactory/tests/DataFactoryElementTests.cs

```diff
@@ -376,38 +503,65 @@ public void SerializationOfDoubleValue()
         [Test]
         public void SerializationOfListOfStringValue()
         {
-            var dfe = new DataFactoryExpression<IList<string>>(ListOfStringValue);
+            var dfe = new DataFactoryElement<IList<string>>(ListOfStringValue);
             var actual = GetSerializedString(dfe);
             Assert.AreEqual(ListOfStringJson, actual);
         }
 
         [Test]
         public void SerializationOfEmptyListOfStringValue()
         {
-            var dfe = new DataFactoryExpression<IList<string>>(EmptyListOfStringValue);
+            var dfe = new DataFactoryElement<IList<string>>(EmptyListOfStringValue);
             var actual = GetSerializedString(dfe);
             Assert.AreEqual(EmptyListJson, actual);
         }
 
         [Test]
         public void SerializationOfExpression()
         {
-            var dfe = DataFactoryExpression<int>.FromExpression(ExpressionValue);
+            var dfe = DataFactoryElement<int>.FromExpression(ExpressionValue);
             var actual = GetSerializedString(dfe);
             Assert.AreEqual(ExpressionJson, actual);
         }
 
+        [Test]
+        public void SerializationOfSecureString()
+        {
+            var dfe = DataFactoryElement<string>.FromLiteral(SecureStringValue, true);
+            var actual = GetSerializedString(dfe);
+            Assert.AreEqual(SecureStringJson, actual);
+            Assert.AreEqual(SecureStringValue, dfe.ToString());
+        }
+
+        [Test]
+        public void SerializationOfSecureStringLiteralNonString()
+        {
+            var dfe = DataFactoryElement<int>.FromLiteral(SecureStringValueNonString, true);
+
+            var actual = GetSerializedString(dfe);
+            Assert.AreEqual(SecureStringJsonNonString, actual);
+            Assert.AreEqual(SecureStringValueNonString.ToString(), dfe.ToString());
+        }
+
+        [Test]
+        public void SerializationOfKeyVaultReference()
+        {
+            var dfe = DataFactoryElement<int>.FromKeyVaultSecretReference(KeyVaultSecretReferenceValue);
```

Comment:
I believe this would influence what the actual resolved value should be. The reference itself is always a string, but it may be resolvable to other types.